org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/RenameHandlerTest.java

```diff
@@ -0,0 +1,221 @@
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.core.internal.handlers;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.eclipse.jdt.core.ICompilationUnit;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IPackageFragment;
+import org.eclipse.jdt.core.IPackageFragmentRoot;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.jdt.ls.core.internal.WorkspaceHelper;
+import org.eclipse.jdt.ls.core.internal.preferences.PreferenceManager;
+import org.eclipse.jdt.ls.core.internal.preferences.Preferences;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.RenameParams;
+import org.eclipse.lsp4j.TextDocumentIdentifier;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.junit.Before;
+import org.junit.Test;
+
+public class RenameHandlerTest extends AbstractCompilationUnitBasedTest {
+
+	private RenameHandler handler;
+
+	private PreferenceManager preferenceManager;
+
+	private IPackageFragmentRoot sourceFolder;
+
+	@Override
+	@Before
+	public void setup() throws Exception {
+		importProjects("eclipse/hello");
+		project = WorkspaceHelper.getProject("hello");
+		IJavaProject javaProject = JavaCore.create(project);
+		sourceFolder = javaProject.getPackageFragmentRoot(javaProject.getProject().getFolder("src"));
+		preferenceManager = mock(PreferenceManager.class);
+		Preferences p = mock(Preferences.class);
+		when(preferenceManager.getPreferences()).thenReturn(p);
+		when(p.isRenameEnabled()).thenReturn(true);
+		handler = new RenameHandler(preferenceManager);
+	}
+
+	@Test
+	public void testRename_parameter() throws JavaModelException {
+		IPackageFragment pack1 = sourceFolder.createPackageFragment("test1", false, null);
+
+		String[] codes = {
+				"package test1;\n",
+				"public class E {\n",
+				"   /** This is a method */\n",
+				"   public int foo(String str) {\n",
+				"  		str|*.length()\n",
+				"   }\n",
+				"   public int bar(String str) {\n",
+				"   	str.length()\n",
+				"   }\n",
+				"}\n"
+		};
+		StringBuilder builder = new StringBuilder();
+		Position pos = mergeCode(builder, codes);
+		ICompilationUnit cu = pack1.createCompilationUnit("E.java", builder.toString(), false, null);
+
+		WorkspaceEdit edit = getRenameEdit(cu, pos, "newname");
+		assertNotNull(edit);
+		assertEquals(edit.getChanges().size(), 1);
+		assertEquals(edit.getChanges().get(JDTUtils.getFileURI(cu)).size(), 2);
+	}

```

Comment:
IMO best is to evaluate the edits (use the `TextEditUtil`) and compare with an expected result. The number of edits seem an implementation detail to me